File: frontend/src/services/project-membership.service.ts
import api from '../api/api';
import type { ProjectMembership, ProjectRole } from '../types/app.types';

export const projectMembershipService = {
  getProjectMembers: async (projectId: string): Promise<ProjectMembership[]> => {
    const response = await api.get(`/api/v1/projects/${projectId}/members`);
    return response.data;
  },

  addProjectMember: async (projectId: string, email: string, role: ProjectRole): Promise<ProjectMembership> => {
    const response = await api.post(`/api/v1/projects/${projectId}/members`, { email, role });
    return response.data;
  },

  updateProjectMemberRole: async (projectId: string, membershipId: string, role: ProjectRole): Promise<ProjectMembership> => {
    const response = await api.put(`/api/v1/projects/${projectId}/members/${membershipId}/role`, { role });
    return response.data;
  },

  removeProjectMember: async (projectId: string, membershipId: string): Promise<void> => {
    await api.delete(`/api/v1/projects/${projectId}/members/${membershipId}`);
  }
};

File: backend/src/main/java/devos/backend/service/ProjectMembershipService.java
package devos.backend.service;

import devos.backend.dto.ProjectMembershipDto;
import devos.backend.model.ProjectMembership;

import java.util.List;
import java.util.UUID;

public interface ProjectMembershipService {
    List<ProjectMembershipDto> getProjectMembers(UUID projectId);
    ProjectMembershipDto addProjectMember(UUID projectId, String userEmail, ProjectMembership.ProjectRole role);
    void removeProjectMember(UUID projectId, UUID membershipId);
    ProjectMembershipDto updateProjectMemberRole(UUID projectId, UUID membershipId, ProjectMembership.ProjectRole role);
}
